src/read.c

```diff
@@ -6699,7 +6699,17 @@ static avifResult avifDecoderApplySampleTransform(const avifDecoder * decoder, a
         return result;
     }
 
-    for (avifBool alpha = AVIF_FALSE; alpha <= decoder->alphaPresent ? AVIF_TRUE : AVIF_FALSE; ++alpha) {
+    for (int i_alpha = 0; i_alpha < 2; ++i_alpha) {
```

Comment:
Could also do something like this:
```c
const int numPasses = decoder->alphaPresent ? 2 : 1;
for (int pass = 0; pass < numPasses; ++pass) {
  avifBool alpha = (pass == 1);
  ...
}
```
Not sure which one's clearer. Feel free to keep yours.

Reply:
Done